spec/jobs/identities_backfill_job_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe IdentitiesBackfillJob, type: :job do
+  describe '#perform' do
+    # create a few rows
+    let!(:deleted) do
+      create(:service_provider_identity, :soft_deleted_5m_ago, :consented)
+    end
+
+    let!(:consented_at_set) do
+      create(:service_provider_identity, :consented)
+    end
+
+    let!(:no_consented_at) do
+      create(:service_provider_identity, :active)
+    end
+
+    subject { described_class.perform_now }
+
+    it 'does not update rows that have been soft-deleted' do
+      expect(deleted.deleted_at).to_not be_nil
+      consent_time = deleted.last_consented_at.to_i
+
+      subject
+      deleted.reload
+
+      expect(deleted.deleted_at).not_to be_nil
+      expect(deleted.last_consented_at.to_i).to eq(consent_time)
```

Comment:
Discussed in Slack, but the `to_i` is necessary here because otherwise we run into weird precision issues, where the DB read vs. in-memory representation seem to have a different resolution and this caused the test to fail over whether or not we show nanoseconds.

The factory sets `deleted_at` to "5 minutes ago" so there will be a distinct difference if the record inadvertently gets updated.